--- cmocka-1.0.0/include/cmocka.h.orig	2015-02-13 13:50:50.000000000 +0100
+++ cmocka-1.0.0/include/cmocka.h	2016-07-01 06:08:43.313760702 +0200
@@ -53,7 +53,7 @@
  */
 
 /* For those who are used to __func__ from gcc. */
-#ifndef __func__
+#if (!defined(__STDC_VERSION__) || (__STDC_VERSION__ < 199901L)) && (!defined(__cplusplus) || (__cplusplus < 201103L)) && !defined(__func__)
 #define __func__ __FUNCTION__
 #endif
 
